packages/frontend/core/src/desktop/pages/workspace/attachment/index.tsx

```diff
@@ -0,0 +1,157 @@
+import { Skeleton } from '@affine/component';
+import {
+  type AttachmentBlockModel,
+  matchFlavours,
+} from '@blocksuite/affine/blocks';
+import {
+  type Doc,
+  DocsService,
+  FrameworkScope,
+  useLiveData,
+  useService,
+} from '@toeverything/infra';
+import { type ReactElement, useLayoutEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { AttachmentViewerView } from '../../../../components/attachment-viewer';
+import { ViewIcon, ViewTitle } from '../../../../modules/workbench';
+import { PageNotFound } from '../../404';
+import * as styles from './index.css';
+
+enum State {
+  Loading,
+  NotFound,
+  Found,
+}
+
+type AttachmentPageProps = {
+  pageId: string;
+  attachmentId: string;
+};
+
+const useLoadAttachment = (pageId: string, attachmentId: string) => {
+  const docsService = useService(DocsService);
+  const docRecord = useLiveData(docsService.list.doc$(pageId));
+
+  const [doc, setDoc] = useState<Doc | null>(null);
+  const [state, setState] = useState(State.Loading);
+  const [model, setModel] = useState<AttachmentBlockModel | null>(null);
+
+  useLayoutEffect(() => {
+    if (!docRecord) {
+      setState(State.NotFound);
+      return;
+    }
+
+    const { doc, release } = docsService.open(pageId);
+    setDoc(doc);
+
+    const disposables: Disposable[] = [];
+    let notFound = true;
+
+    if (doc.blockSuiteDoc.ready) {
+      const block = doc.blockSuiteDoc.getBlock(attachmentId);
+      if (block) {
+        notFound = false;
+        setModel(block.model as AttachmentBlockModel);
+        setState(State.Found);
+      }
+    }
+
+    if (notFound) {
+      doc.blockSuiteDoc.load();
+
+      const tid = setTimeout(() => setState(State.NotFound), 5 * 10000); // 50s
```

Comment:
watchUntilLoad? @EYHN

Reply:
could add `waitForSyncd` to doc entity same as `waitForSyncReady`
in packages/common/infra/src/modules/doc/entities/doc.ts